bson/bson.go

```diff
@@ -15,7 +16,7 @@ func GenerateCursorQuery(shouldSecondarySortOnID bool, paginatedField string, co
 		query = map[string]interface{}{"$or": []map[string]interface{}{
 			{paginatedField: map[string]interface{}{comparisonOp: cursorFieldValues[0]}},
 			{"$and": []map[string]interface{}{
-				{paginatedField: map[string]interface{}{"$eq": cursorFieldValues[0]}},
+				{paginatedField: map[string]interface{}{fmt.Sprintf("%se", comparisonOp): cursorFieldValues[0]}},
```

Comment:
```suggestion
				{paginatedField: map[string]interface{}{rangeOperator: cursorFieldValues[0]}},
```
We could make this more readable by creating the variable first